build.rs

```diff
@@ -1,23 +1,26 @@
 use std::env;
 
-include!("src/dist/triple.rs");
+use platforms::Platform;
 
-fn from_build() -> Result<PartialTargetTriple, String> {
-    PartialTargetTriple::new(&triple).ok_or(triple)
+fn from_build() -> Result<String, String> {
     let triple =
         env::var("RUSTUP_OVERRIDE_BUILD_TRIPLE").unwrap_or_else(|_| env::var("TARGET").unwrap());
+    if Platform::ALL.iter().any(|p| p.target_triple == triple) {
+        Ok(triple)
+    } else {
+        Err(triple)
+    }
 }
 
 fn main() {
     println!("cargo:rerun-if-env-changed=RUSTUP_OVERRIDE_BUILD_TRIPLE");
     println!("cargo:rerun-if-env-changed=TARGET");
     match from_build() {
-        Ok(triple) => eprintln!("Computed build based partial target triple: {triple:#?}"),
+        Ok(triple) => eprintln!("Computed build based on target triple: {triple:#?}"),
         Err(s) => {
-            eprintln!("Unable to parse target '{s}' as a PartialTargetTriple");
+            eprintln!("Unable to parse target '{s}' as an existing target triple");
             eprintln!(
-                "If you are attempting to bootstrap a new target you may need to adjust the\n\
-               permitted values found in src/dist/triple.rs"
+                "If you are attempting to bootstrap a new target, you might need to update `platforms` to a newer version in `Cargo.toml`"
```

Comment:
Leave off "to a newer version in `Cargo.toml`"? A `Cargo.lock` update might be sufficient and I don't think the added detail is necessary here.

Do you know/understand why we need to emit the `rustc-env` value?

Reply:
> Leave off "to a newer version in `Cargo.toml`"? A `Cargo.lock` update might be sufficient and I don't think the added detail is necessary here.
Done.

> Do you know/understand why we need to emit the `rustc-env` value?
No idea, introduced in https://github.com/rust-lang/rustup/pull/1704 back in 2019 though. I highly suspect this has something to do with the current CI: 

https://github.com/search?q=repo%3Arust-lang%2Frustup+TARGET+language%3Ayaml&type=code

Reply:
> Leave off "to a newer version in `Cargo.toml`"?
> A `Cargo.lock` update might be sufficient and I don't think the added detail is necessary here.

@djc Done.

> Do you know/understand why we need to emit the `rustc-env` value?

This is required by a potential platform-specific `miri` use. See https://github.com/rust-lang/rust/blob/804421dff5542c9c7da5c60257b5dbc849719505/src/tools/miri/build.rs#L4-L7.